lib/mix/lib/mix/tasks/archive.install.ex

```diff
@@ -27,26 +40,42 @@ defmodule Mix.Tasks.Archive.Install do
     * `--force` - forces installation without a shell prompt; primarily
       intended for automation in build systems like `make`
 
+    * `--submodules` - fetch repository submodules before building archive from
+      git or github
+
+    * `--app` - specify a custom app name to be used for building the archive
+      from git, github, or hex
   """
 
   @behaviour Mix.Local.Installer
 
-  @switches [force: :boolean, sha512: :string]
+  @switches [force: :boolean, sha512: :string, submodules: :boolean, app: :string]
   @spec run(OptionParser.argv) :: boolean
   def run(argv) do
     Mix.Local.Installer.install({__MODULE__, :archive}, argv, @switches)
   end
 
   ### Mix.Local.Installer callbacks
 
-  def check_path_or_url(path_or_url) do
+  def check_install_spec({:fetcher, _}, opts) do
```

Comment:
Can't we do this check inside the installer since it applies to both archive and escript?